tools/export/gnuarmeclipse/__init__.py

```diff
@@ -0,0 +1,891 @@
+        self.options['c'] = {}
+        self.options['cpp'] = {}
+        self.options['ld'] = {}
+
+        libraries = []
+        # print 'libraries'
+        # print self.resources.libraries
+        for lib in self.resources.libraries:
+            l, _ = splitext(basename(lib))
+            libraries.append(l[3:])
+
+        self.system_libraries = [
+            'stdc++', 'supc++', 'm', 'c', 'gcc', 'nosys'
+        ]
+
+        # TODO: get the list from existing .cproject
+        build_folders = ['Debug', 'Release', 'BUILD']
+
+        objects = [self.filter_dot(s) for s in self.resources.objects]
+        for bf in build_folders:
+            objects = [o for o in objects if not o.startswith(bf + '/')]
+        # print 'objects'
+        # print objects
+
+        self.compute_exclusions()
+
+        self.process_options()
+
+        self.options['as']['defines'] = self.toolchain.get_symbols(True)
+        self.options['c']['defines'] = self.toolchain.get_symbols()
+        self.options['cpp']['defines'] = self.toolchain.get_symbols()
+        print 'Symbols: {0}'.format(len(self.options['cpp']['defines']))
+
+        self.options['common']['include_paths'] = [
+            self.filter_dot(s) for s in self.resources.inc_dirs]
+        print 'Include folders: {0}'.format(len(self.options['common']['include_paths']))
+        self.options['common']['excluded_folders'] = '|'.join(
+            self.excluded_folders)
+
+        self.options['ld']['library_paths'] = [
+            self.filter_dot(s) for s in self.resources.lib_dirs]
+
+        self.options['ld']['object_files'] = objects
+
+        self.options['ld']['user_libraries'] = libraries
+
+        self.options['ld']['system_libraries'] = self.system_libraries
+
+        self.options['ld']['script'] = self.filter_dot(
+            self.resources.linker_script)
+        print 'Linker script: {0}'.format(self.options['ld']['script'])
+
+        ctx = {
+            'name': self.project_name,
+
+            # Compiler & linker command line options
+            'options': self.options,
+
+            # Unique IDs used in multiple places.
+            # Those used only once are implemented with {{u.id}}.
+            'debug_config_uid': u.id,
+            'debug_tool_c_compiler_uid': u.id,
+            'debug_tool_c_compiler_input_uid': u.id,
+            'debug_tool_cpp_compiler_uid': u.id,
+            'debug_tool_cpp_compiler_input_uid': u.id,
+            'release_config_uid': u.id,
+            'release_tool_c_compiler_uid': u.id,
+            'release_tool_c_compiler_input_uid': u.id,
+            'release_tool_cpp_compiler_uid': u.id,
+            'release_tool_cpp_compiler_input_uid': u.id,
+
+            # Must be an object with an `id` property, which
+            # will be called repeatedly, to generate multiple UIDs.
+            'u': u,
+        }
+
+        # TODO: it would be good to have jinja stop if one of the
+        # expected context values is not defined.
```

Comment:
Many thanks. I may tack on a few commits after your changes in that PR to ensure that all of the exporters will work afterwards.